src/geo/transform.js

```diff
@@ -492,41 +459,43 @@ class Transform {
 
         if (this.latRange) {
             const latRange = this.latRange;
-            minY = this.latY(latRange[1]);
-            maxY = this.latY(latRange[0]);
+            minY = this.project(new LngLat(0, latRange[1])).y;
+            maxY = this.project(new LngLat(0, latRange[0])).y;
             sy = maxY - minY < size.y ? size.y / (maxY - minY) : 0;
         }
 
         if (this.lngRange) {
             const lngRange = this.lngRange;
-            minX = this.lngX(lngRange[0]);
-            maxX = this.lngX(lngRange[1]);
+            minX = this.project(new LngLat(lngRange[0], 0)).x;
+            maxX = this.project(new LngLat(lngRange[1], 0)).x;
```

Comment:
To add to my point above, here, instead of zero allocations, we do 4 * (1 + 3) = 16, along with double-projection.